### DISCUSSION TITLE
MotherDuck talk by Jordan Tigani: contradiction in definition of Big Data

### DISCUSSION THREAD

**Discussion by oliverangelil** [duckdb/duckdb#14861] [General]
I watched [this](https://youtu.be/VIK_zgio4vk?si=4S26cjvVKaXNKMst) video by Jordan Tigani

At 2:50 he defines big data as "Anything that is too big to run on a single machine"
Then at 19:00, it's anything greater than 1TB...

Memory in a compute node ("1 machine") is typically around 16GB... 
So the 2nd definition is confusing me...

**Comment by carlopi** [comment]
The laptop where I am typing this has 994,66 GB of memory, that's not too short of 1 TB.

**Reply by oliverangelil** [reply]
Oh... my macbook has 16gb memory...
I think 1TB is not the norm at all.

**Reply by oliverangelil** [reply]
I see there are some Azure VMs that go up to 32 TB ram...

Good to know - this really raises the data processing limit for duckDB 😍

**Reply by carlopi** [reply]
I expressed myself wrongly.

There are two limiting factors, RAM and local available storage.
16 GB / about 1 TB in my case.

There are a few it-depends, where on certain operators limiting factors will be RAM, but a common case is being limited by available local storage.

Your mileage varies with queries, https://duckdb.org/2024/03/29/external-aggregation.html is a good start.
